tests/ignite/conftest.py

```diff
@@ -333,3 +336,159 @@ def _gloo_hvd_execute(func, args, np=1, do_init=False):
 @pytest.fixture()
 def gloo_hvd_executor():
     yield _gloo_hvd_execute
+
+
+skip_if_has_not_native_dist_support = pytest.mark.skipif(
+    not idist.has_native_dist_support, reason="Skip if no native dist support"
+)
+skip_if_has_not_xla_support = pytest.mark.skipif(not idist.has_xla_support, reason="Skip if no PyTorch XLA package")
+skip_if_has_not_horovod_support = pytest.mark.skipif(
+    not idist.has_hvd_support, reason="Skip if no Horovod dist support"
+)
+
+
+is_horovod_stash_key = pytest.StashKey[bool]()
+is_xla_stash_key = pytest.StashKey[bool]()
+is_xla_single_device_stash_key = pytest.StashKey[bool]()
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        pytest.param(
+            "nccl_gpu",
+            marks=[
+                pytest.mark.distributed,
+                skip_if_has_not_native_dist_support,
+                pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU"),
+            ],
+        ),
+        pytest.param("gloo_cpu_or_gpu", marks=[pytest.mark.distributed, skip_if_has_not_native_dist_support]),
+        pytest.param(
+            "horovod",
+            marks=[
+                pytest.mark.distributed,
+                skip_if_has_not_horovod_support,
+                pytest.mark.skipif("WORLD_SIZE" in os.environ, reason="Skip if launched as multiproc"),
+            ],
+        ),
+        pytest.param(
+            "multinode_gloo_cpu_or_gpu",
+            marks=[
+                pytest.mark.multinode_distributed,
+                skip_if_has_not_native_dist_support,
+                pytest.mark.skipif("MULTINODE_DISTRIB" not in os.environ, reason="Skip if not multi-node distributed"),
+            ],
+        ),
+        pytest.param(
+            "multinode_nccl_gpu",
+            marks=[
+                pytest.mark.multinode_distributed,
+                skip_if_has_not_native_dist_support,
+                pytest.mark.skipif(
+                    "GPU_MULTINODE_DISTRIB" not in os.environ, reason="Skip if not multi-node distributed"
+                ),
+            ],
+        ),
+        pytest.param(
+            "single_device_xla",
+            marks=[
+                pytest.mark.tpu,
+                skip_if_has_not_xla_support,
+                pytest.mark.skipif("NUM_TPU_WORKERS" in os.environ, reason="Skip if NUM_TPU_WORKERS is in env vars"),
+            ],
+        ),
+        pytest.param(
+            "xla_nprocs",
+            marks=[
+                pytest.mark.tpu,
+                skip_if_has_not_xla_support,
+                pytest.mark.skipif(
+                    "NUM_TPU_WORKERS" not in os.environ, reason="Skip if no NUM_TPU_WORKERS in env vars"
+                ),
+            ],
+        ),
+    ],
+)
+def distributed(request, local_rank, world_size):
+    if request.param == "nccl_gpu" or request.param == "gloo_cpu_or_gpu":
```

Comment:
Let's code such condition as 
```suggestion
    if request.param in ("nccl_gpu", "gloo_cpu_or_gpu"):
```